packages/backend/src/repositories/product.ts

```diff
@@ -29,8 +32,35 @@ export class ProductRepository {
       where: {
         type,
         categoryId,
+        status: ProductStatus.ACTIVE,
+        price: {
+          gt: +priceGt,
+          lte: +priceLt,
+        },
+      },
+    });
+  }
+
+  public async getAllItemsLength(query: ProductQuery) {
+    const {
+      type,
+      categoryId,
+      priceGt = ITEM_FILTER.PRICE_GT_DEFAULT,
+      priceLt = ITEM_FILTER.PRICE_LT_DEFAULT,
+    } = query;
+    const items = await this._dbClient.product.findMany({
+      where: {
+        type,
+        categoryId,
+        status: ProductStatus.ACTIVE,
+        price: {
+          gt: +priceGt,
+          lte: +priceLt,
+        },
       },
     });
+
+    return items.length;
   }
```

Comment:
1. create object with filters and order condition.
2. create 2 queries: select + pagination, and count
3. promise.all(query1, query2)